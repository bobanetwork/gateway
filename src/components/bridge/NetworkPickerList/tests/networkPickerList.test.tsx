import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import CustomThemeProvider from 'themes'
import { NetworkList } from '../index'

const mockStore = configureStore()

const renderNetworkList = ({ options = null }: any) => {
  return render(
    <Provider
      store={mockStore({
        ui: {
          theme: 'dark',
        },
        network: {
          activeNetworkType: 'Testnet',
        },
        networkType: 'Testnet',
        layer: 'L2',
        bridge: {
          bridgeType: 'Light',
        },
        setup: {
          accountEnabled: false,
          netLayer: 'L2',
          baseEnabled: false,
          walletAddress: '0x1e2855A0EA33d5f293E5Ba2018874FAB9a7F05B3',
        },
        ...options,
      })}
    >
      <CustomThemeProvider>
        <NetworkList close={() => null} isIndependentDestNetwork={true} />
      </CustomThemeProvider>
    </Provider>
  )
}

describe('NetworkList Component', () => {
  it('renders NetworkList correctly', () => {
    const { asFragment } = renderNetworkList({})
    expect(asFragment).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
})
