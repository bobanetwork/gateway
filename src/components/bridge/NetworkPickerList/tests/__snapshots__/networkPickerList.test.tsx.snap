// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkList Component renders NetworkList correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-ikkxIA jVsqAY"
  >
    <div
      class="sc-dAbbOL hKpqkV networkItem"
    >
      <div
        class="sc-feUZmu dYqRNZ"
      >
        <svg
          fill="none"
          viewBox="0 0 1202 1202"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="601"
            cy="601"
            fill="white"
            r="601"
          />
          <path
            d="M682 711.3H375.9V264H682C787.1 264 872.2 349.2 872.2 454.2V521C872.2 626.1 787.1 711.3 682 711.3Z"
            fill="#ACD905"
          />
          <path
            d="M760.9 935H375.9V711.3C375.9 587.8 476 487.7 599.5 487.7H760.9C884.4 487.7 984.5 587.8 984.5 711.3C984.5 834.8 884.4 935 760.9 935Z"
            fill="#AEAFAA"
          />
          <path
            d="M872.2 517.4V521.1C872.2 626.2 787.1 711.3 681.9 711.3H375.9C375.9 649.6 401 593.7 441.4 553.2C481.9 512.7 537.8 487.7 599.6 487.7H760.9C801.5 487.7 839.5 498.5 872.2 517.4Z"
            fill="#5D6E3C"
          />
          <path
            d="M376.5 824C500.488 824 601 723.488 601 599.5C601 475.512 500.488 375 376.5 375C252.512 375 152 475.512 152 599.5C152 723.488 252.512 824 376.5 824Z"
            fill="#627EEA"
          />
          <path
            d="M376.5 403.2L373.9 412.2V671.7L376.5 674.4L497 603.1L376.5 403.2Z"
            fill="#C5C5C5"
          />
          <path
            d="M376.5 403.2L256 603.1L376.5 674.4V548.4V403.2Z"
            fill="#E7E7E7"
          />
          <path
            d="M376.5 697.2L375 699V791.4L376.5 795.8L497.1 626L376.5 697.2Z"
            fill="#C5C5C5"
          />
          <path
            d="M376.5 795.8V697.2L256 626L376.5 795.8Z"
            fill="#E7E7E7"
          />
          <path
            d="M376.5 674.4L497 603.1L376.5 548.4V674.4Z"
            fill="#909090"
          />
          <path
            d="M256 603.1L376.5 674.4V548.4L256 603.1Z"
            fill="#AEAEAE"
          />
        </svg>
      </div>
      <p
        class="sc-dhKdcB ibpgnY sc-fUnMCh hFwnBZ"
      >
        Boba (Sepolia)
      </p>
    </div>
    <div
      class="sc-dAbbOL hKpqkV networkItem"
    >
      <div
        class="sc-feUZmu dYqRNZ"
      >
        <svg
          fill="none"
          viewBox="0 0 1202 1202"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="601"
            cy="601"
            fill="white"
            r="601"
          />
          <path
            d="M682 712.3H375.9V265H682C787.1 265 872.2 350.2 872.2 455.2V522C872.2 627.1 787.1 712.3 682 712.3Z"
            fill="#ACD905"
          />
          <path
            d="M760.9 936H375.9V712.3C375.9 588.8 476 488.7 599.5 488.7H760.9C884.4 488.7 984.5 588.8 984.5 712.3C984.5 835.8 884.4 936 760.9 936Z"
            fill="#AEAFAA"
          />
          <path
            d="M872.2 518.4V522.1C872.2 627.2 787.1 712.3 681.9 712.3H375.9C375.9 650.6 401 594.7 441.4 554.2C481.9 513.7 537.8 488.7 599.6 488.7H760.9C801.5 488.7 839.5 499.5 872.2 518.4Z"
            fill="#5D6E3C"
          />
          <path
            d="M376.5 825C500.488 825 601 724.488 601 600.5C601 476.512 500.488 376 376.5 376C252.512 376 152 476.512 152 600.5C152 724.488 252.512 825 376.5 825Z"
            fill="black"
          />
          <path
            d="M412.4 723.3V765L376.1 786.4L340.8 765V723.3L376.1 744.7L412.4 723.3ZM217.9 579.1L253.2 600.5V672L314 708.3V750L217.9 693.3V579.1ZM534.1 579.1V693.5L437 750.2V708.5L497.8 672.2V600.7L534.1 579.1ZM436.8 522.5L473.1 543.9V585.5L412.3 621.8V694.3L377 715.7L341.9 694.3V621.6L278.9 585.3V543.6L315.2 522.2L376 558.5L436.8 522.5ZM278.9 615.3L314.2 636.7V678.4L278.9 657V615.3ZM473.2 615.3V657L437.9 678.4V636.7L473.2 615.3ZM253.2 486.1L289.5 507.5L253.2 528.9V570.5L217.9 549.1V507.5L253.2 486.1ZM498.8 486.1L535.1 507.5V549.2L498.8 570.5V528.8L463.5 507.5L498.8 486.1ZM376 486.1L412.3 507.5L376 528.9L340.7 507.5L376 486.1ZM376 414.7L473.1 471.4L437.8 492.8L377 456.5L315 492.8L279.7 471.4L376 414.7Z"
            fill="#F1B82F"
          />
        </svg>
      </div>
      <p
        class="sc-dhKdcB ibpgnY sc-fUnMCh hFwnBZ"
      >
        Boba BNB Testnet
      </p>
    </div>
  </div>
</DocumentFragment>
`;
